cmake/onnxruntime_training.cmake

```diff
@@ -47,9 +47,9 @@ add_dependencies(onnxruntime_training_runner ${onnxruntime_EXTERNAL_DEPENDENCIES
 onnxruntime_add_include_to_target(onnxruntime_training_runner onnxruntime_common onnx onnx_proto protobuf::libprotobuf onnxruntime_training)
 
 if (onnxruntime_USE_CUDA)
-  target_include_directories(onnxruntime_training_runner PRIVATE ${ONNXRUNTIME_ROOT} ${ORTTRAINING_ROOT} ${eigen_INCLUDE_DIRS} PUBLIC ${onnxruntime_graph_header} ${onnxruntime_CUDNN_HOME}/include ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+  target_include_directories(onnxruntime_training_runner PRIVATE ${ONNXRUNTIME_ROOT} ${ORTTRAINING_ROOT} ${eigen_INCLUDE_DIRS} PUBLIC ${onnxruntime_graph_header} ${onnxruntime_CUDNN_HOME}/include ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}  ${PROJECT_SOURCE_DIR}/external/json)
```

Comment:
these lines are kind of long.
should be able to call target_include_directories() multiple times on a single target with the effect of appending additional includes. so something like this would be more concise:
```
target_include_directories(onnxruntime_training_runner ... ${PROJECT_SOURCE_DIR}/external/json)
if (onnxruntime_USE_CUDA)
  target_include_directories(onnxruntime_training_runner ...)
endif()
```
also, can the json include be PRIVATE? it's only used in a .cc file.